rime_main.c

```diff
@@ -64,6 +64,112 @@ static void notification_handler(void* context_object,
   }
 }
 
+static int n_plugin_handles;
+static void **plugin_handles;
+static const char **plugin_modules;
+
+static int load_plugins(const char *config_file)
+{
+  RimeConfig config = {0};
+  void **new_plugin_handles;
+
+  if (!rime_api->config_open(config_file, &config)) {
+    g_error("error loading settings for %s\n", config_file);
+    return 1;
+  }
+
+  RimeConfigIterator iter;
+  int n = 0;
+  if (rime_api->config_begin_list(&iter, &config, "plugins")) {
+    while(rime_api->config_next(&iter)) n++;
+    rime_api->config_end(&iter);
+  }
+
+  int m = 0;
+  if (rime_api->config_begin_list(&iter, &config, "modules")) {
+    while(rime_api->config_next(&iter)) m++;
+    rime_api->config_end(&iter);
+  }
+
+  // reserve space new plugins
+  new_plugin_handles = realloc(plugin_handles, sizeof(void *) * (n_plugin_handles + n));
+  if (!new_plugin_handles) {
+    return 2;
+  } else {
+    plugin_handles = new_plugin_handles;
+  }
+
+  plugin_modules = malloc(sizeof(const char *) * (m + 2));
```

Comment:
free memory first, including allocated strings.

Reply:
No. `plugin_modules` should be `NULL` at the point.
There are only two cases:
1. `main()` -> `rime_with_ibus()` -> `ibus_rime_start()`
2. `ibus_rime_engine_property_activate()`(deploy) -> `ibus_rime_stop()`(memory freed) -> `ibus_rime_start()`